src/generate/register.rs

```diff
@@ -652,19 +656,80 @@ pub fn fields(
             }
 
             if width == 1 {
-                proxy_items.extend(quote! {
-                    ///Sets the field bit
-                    #inline
-                    pub #unsafety fn set_bit(self) -> &'a mut W {
-                        self.bit(true)
-                    }
+                if mwv != ModifiedWriteValues::Modify {
+                    unsafety = Some(Ident::new("unsafe", Span::call_site()));
```

Comment:
What's the purpose of this? It seems like it just makes `bits()` for this field `unsafe` when it's not `Modify`, but that won't affect people calling `update` and also doesn't affect the `set_bit()` etc methods, right?

Reply:
What is `update`?

Reply:
Sorry, I meant `modify()` ([talking about this](https://github.com/rust-embedded/svd2rust/issues/540#issuecomment-934580846)).

You're right, it will only affect `fn bit(bool)`. I'm not sure about making `bit()` unsafe though, since it's still safe to call and you can always write one of the other methods to get the same effect, which is why we make `bits()` safe if there's an enumeratedValue for every possible value. What would the safety requirements on `bit()` be, since you can safely call the other method to set it true or false?

Reply:
I've not thought about this. But if incorrect bit just do nothing we could make `bit()` safe everywhere.